sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -248,31 +250,43 @@ export class ManagementClient extends LinkEntity<RequestResponseLink> {
         abortSignal
       );
 
-      this.link!.sender.on(SenderEvents.senderError, (context: EventContext) => {
-        const id = context.connection.options.id;
-        const ehError = translate(context.sender!.error!);
-        logError(
-          ehError,
-          "[%s] An error occurred on the $management sender link.. %O",
-          id,
-          ehError
-        );
-      });
-      this.link!.receiver.on(ReceiverEvents.receiverError, (context: EventContext) => {
-        const id = context.connection.options.id;
-        const ehError = translate(context.receiver!.error!);
-        logError(
-          ehError,
-          "[%s] An error occurred on the $management receiver link.. %O",
-          id,
-          ehError
-        );
-      });
+      // Attach listeners for the `sender_error` and `receiver_error` events to log the errors.
+      //  - It is possible that the previous "_init" call had already added the listeners
+      //    (example: parallel _init calls can cause this),
+      //    hence checking the count of the listeners and adding them only if they're not present.
+      const senderErrorListenerCount = this.link?.sender.listenerCount(SenderEvents.senderError);
+      const receiverErrorListenerCount = this.link?.receiver.listenerCount(
+        ReceiverEvents.receiverError
+      );
+      if (senderErrorListenerCount && senderErrorListenerCount < 1) {
+        this.link!.sender.on(SenderEvents.senderError, (context: EventContext) => {
```

Comment:
There are options that you can pass to initLink for callbacks that are intended to be only set once - if the ones you're setting aren't there we should add them and register them there.

That should eliminate the need to do this kind of checking.

Reply:
I did look into that. There are complications, however.

"rhea" doesn't allow setting only the "onError" handler in the options if it is not accompanied by an "onMessage" handler.
![image](https://user-images.githubusercontent.com/10452642/95610179-bbb8b480-0a14-11eb-9f04-1dc9a59dcb83.png)

Our `onMessage` handler("message" event) is set at core-amqp and we are trying to pass the `onError` handler in the options.
If we want to pass both `onError` and `onMessage` through core-amqp, we'd need to update the core-amqp's logic drastically, which would also impact event-hubs I guess.

Reply:
You have the lock to protect you here so you can do one-time init-only modification to the receiver.

Reply:
Good idea, updated createRheaLink instead to add the listener and thus, this would happen as a one-time init-only modification.